src/Prefix.jl

```diff
@@ -475,3 +475,33 @@ function cleanup_dependencies(prefix::Prefix, artifact_paths)
         unsymlink_tree(art_path, joinpath(prefix, "destdir"))
     end
 end
+
+"""
+    compress_dir(dir::AbstractString;
+                 compression_stream = GzipCompressorStream,
+                 extension::AbstractString = ".gz",
+                 verbose::Bool = false)
+
+Compress all files in `dir` using the specified `compression_stream`, appending
+`extension` to the filenames and remove the original uncompressed files.
+"""
+function compress_dir(dir::AbstractString;
+                      compression_stream = GzipCompressorStream,
+                      extension::AbstractString = ".gz",
+                      verbose::Bool = false)
+    if isdir(dir)
+        if verbose
+            @info "Compressing files in $(dir)"
+        end
+        for f in readdir(dir)
+            filename = joinpath(dir, f)
+            if isfile(filename)
+                text = read(filename, String)
+                open(compression_stream, filename * extension, "w") do stream
```

Comment:
I haven't figured out how to do this with the `open() do` form yet, but we should probably compress with level 9.

Reply:
Done!  For the OpenBLAS example [above](https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/755#issuecomment-616135648) there isn't a large improvement though:
```
shell> ls -lh
total 336K
-rw-r--r-- 1 mose mose 331K Apr 20 22:34 OpenBLAS.log.gz
-rw-r--r-- 1 mose mose  165 Apr 20 22:34 relink_to_rpath_libopenblas64_.0.3.9.so.log.gz
```

Reply:
Every little bit helps.  :)